### ISSUE TITLE
Use pre-trained Calamari models

### ISSUE THREAD

**Issue by zhangxiangnick** [Calamari-OCR/calamari#5]
Thanks for the great work!

I installed Calamari on a new AWS P2 instance and [calamari-models](https://github.com/Calamari-OCR/calamari_models/tree/master/default). Tried to test on a simple [example](http://www.robots.ox.ac.uk/~vgg/data/text/data.png) by

```
calamari-predict --checkpoint calamari_models/default/ModernEnglish.ckpt --files data.png
``` 

The detected text is way off. I guess it is related to the loading of model. 

I got these warnings:
```
Found 1 files in the dataset
2018-08-05 17:12:16.976735: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Attempting a workaround: New graph and load weights
Using CUDNN compatible LSTM backend on CPU
WARNING:tensorflow:From /home/ubuntu/anaconda3/envs/calamari/lib/python3.6/site-packages/tensorflow/python/ops/rnn.py:417: calling reverse_sequence (from tensorflow.python.ops.array_ops) with seq_dim is deprecated and will be removed in a future version.
Instructions for updating:
seq_dim is deprecated, use seq_axis instead
WARNING:tensorflow:From /home/ubuntu/anaconda3/envs/calamari/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py:432: calling reverse_sequence (from tensorflow.python.ops.array_ops) with batch_dim is deprecated and will be removed in a future version.
Instructions for updating:
batch_dim is deprecated, use batch_axis instead
2018-08-05 17:12:20.637472: W tensorflow/core/framework/op_kernel.cc:1318] OP_REQUIRES failed at save_restore_v2_ops.cc:184 : Not found: Key Minimum/ExponentialMovingAverage not found in checkpoint
Attempting workaround: only loading trainable variables
Loading Dataset: 100%|█████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 109.32it/s]
Data Preprocessing: 100%|██████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 104.47it/s]
Prediction: 100%|███████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  7.74it/s]
Prediction of 1 models took 0.14934062957763672s
```

Is it due to the tensorflow version that ExponentialMovingAverage are not loaded? Currently installing calamari will install tensowflow 1.9. What tf version do you use in your development?

Thanks!

**Comment by ChWick** [comment]
Some calamari models in the repository are not yet trained with the moving average (older calamari version), however this has no effect on the prediction.
The deprecated warnings will be 'fixed' in a future update, to fix "TensorFlow binary was not compiled to use: AVX2 FMA" you need to compile Tensorflow for your machine.

**Comment by alwc** [comment]
@ChWick I've moved all the test `.png`  and `.gt.txt` into `test-calamari` and ran the following code:

```
calamari-predict --checkpoint calamari_models/default/ModernEnglish.ckpt --files /test-calamari/*.png
```

Next, I ran

```
calamari-eval --gt data/test-calamari/*.gt.txt
```

and here are the result:

```
Loading Dataset: 100% 9730/9730 [00:00<00:00, 40378.65it/s]
Text Preprocessing: 100% 9730/9730 [00:01<00:00, 6694.55it/s]
Loading Dataset: 100% 9730/9730 [00:00<00:00, 43940.55it/s]
Text Preprocessing: 100% 9730/9730 [00:01<00:00, 7067.60it/s]
Evaluation: 100% 9730/9730 [00:03<00:00, 2773.32it/s]
Evaluation result
=================

Got mean normalized label error rate of 0.43% (1957 errs, 459429 total chars, 1973 sync errs)
GT       PRED     COUNT    PERCENT   
{ }      {}             76      3.85%
{}       { }            55      2.79%
{.}      {,}            48      2.43%
{,}      {.}            42      2.13%
{'}      {}             31      1.57%
{e}      {c}            29      1.47%
{O}      {0}            26      1.32%
{i}      {l}            19      0.96%
{1}      {l}            17      0.86%
{}       {.}            15      0.76%
The remaining but hidden errors make up 81.86%
```

Should I get normalized label error rate close to 0.155% (as indicated in the paper) if I'm using the `ModernEnglish.ckpt` pretrained weights? Thanks.

**Comment by alwc** [comment]
@ChWick Thanks for your help! So I updated `calamari_ocr` to `0.1.8` and downloaded the new pretrained models. Here were the steps I did afterward:

```
$: calamari-predict --checkpoint calamari_models/default/ModernEnglish.ckpt --files data/test-calamari/*.png

Found 9730 files in the dataset
2018-08-07 02:54:08.754899: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Using CUDNN compatible LSTM backend on CPU
WARNING:tensorflow:From /usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/rnn.py:417: calling reverse_sequence (from tensorflow.python.ops.array_ops) with seq_dim is deprecated and will be removed in a future version.
Instructions for updating:
seq_dim is deprecated, use seq_axis instead
WARNING:tensorflow:From /usr/local/lib/python3.5/dist-packages/tensorflow/python/util/deprecation.py:432: calling reverse_sequence (from tensorflow.python.ops.array_ops) with batch_dim is deprecated and will be removed in a future version.
Instructions for updating:
batch_dim is deprecated, use batch_axis instead
Loading Dataset: 100%|##########################################################################################################################| 9730/9730 [00:13<00:00, 737.25it/s]
Data Preprocessing: 100%|########################################################################################################################| 9730/9730 [02:27<00:00, 65.76it/s]
Prediction: 100%|################################################################################################################################| 9730/9730 [29:41<00:00,  5.46it/s]
Prediction of 1 models took 1942.9190566539764s
All files written
```
Next, I ran the evaluation:

```
$: calamari-eval --gt data/test-calamari/*.gt.txt --checkpoint calamari_models/default/ModernEnglish.ckpt                         
                        
Resolving files
Loading Dataset: 100%|########################################################################################################################| 9730/9730 [00:00<00:00, 40278.74it/s]
Text Preprocessing: 100%|#####################################################################################################################| 9730/9730 [00:00<00:00, 13625.54it/s]
Loading Dataset: 100%|########################################################################################################################| 9730/9730 [00:00<00:00, 32110.62it/s]
Text Preprocessing: 100%|#####################################################################################################################| 9730/9730 [00:00<00:00, 13351.66it/s]
Evaluation: 100%|##############################################################################################################################| 9730/9730 [00:02<00:00, 4504.07it/s]
Evaluation result
=================

Got mean normalized label error rate of 0.23% (1078 errs, 459464 total chars, 1082 sync errs)
GT       PRED     COUNT    PERCENT   
{,}      {.}            33      3.05%
{}       { }            32      2.96%
{ }      {}             31      2.87%
{.}      {,}            18      1.66%
{i}      {l}            11      1.02%
{O}      {0}            10      0.92%
{n}      {}              9      0.83%
{'}      {}              9      0.83%
{y}      {v}             9      0.83%
{.}      {}              8      0.74%
The remaining but hidden errors make up 84.29%
```

The result has improved, but I still couldn't match the CER as indicated in the paper. Did I miss anything in my procedure?

**Comment by 18813055625** [comment]
@zhangxiangnick  /ModernEnglish.ckpt在哪里下载？

**Comment by ChWick** [comment]
@alwc Add --n_confusions -1 to calamari-eval and you should find a few lines of wrong GT (empty prediction but full string as GT) that should make up approx 25% of the remaining discrepancy.

**Comment by alwc** [comment]
@ChWick So the calamari number 0.155% in Table 2 is obtained by "0.23 * (1 - sum of wrong GT %) = 0.155"?

BTW, this prediction is really weird:

```
GT        PRED                                                                                                        COUNT PERCENT
{49}     {Foam Rubber Modeling of El Centro Terminal Substation Building}        1      5.73%
```

@18813055625 https://github.com/Calamari-OCR/calamari_models

**Comment by ChWick** [comment]
I manually detected those lines and rotated them (all 3? lines are rotated by 90 degrees). To find them you can use the --xlsx_output flag of the eval script. Sort for highest relative error and rotate the corresponding lines.

**Comment by MrOCR** [comment]
@zhangxiangnick According to the sample example `data.png` that you provided, your detected text is way off thats because the `ModernEnglish.ckpt` model is not trained on similar data.
Instead, it can recognize modern english text without augmentation, example:
![11](https://user-images.githubusercontent.com/33478216/46499779-a909b480-c829-11e8-87f2-d4a34d84ab69.png)
If you want Calamari to recognize images similar to the example you provided, you should train a new model to do so, by providing training data/ images that are similar to what you wan to recognize.

**Comment by MrOCR** [comment]
@alwc It's clear that you don't know what your talking about.
The images that you are testing on, are just a small sample of 50 lines from the UW3 dataset.

@ChWick Please close this Issue.
